### PR TITLE
gh-106751: selectors: optimize EpollSelector.select()

### PR DESCRIPTION
**Pull request by bdraco**
* Issue: gh-106751

### CODE REVIEW

File: Lib/selectors.py [hunk 1]
```diff
@@ -430,6 +430,9 @@ class PollSelector(_PollLikeSelector):
 
 if hasattr(select, 'epoll'):
 
+    NOT_EPOLLIN = ~select.EPOLLIN
```
Comment (eendebakpt) [inline Lib/selectors.py]: Perhaps make this private, since it is in the module namespace.
```suggestion
    _NOT_EPOLLIN = ~select.EPOLLIN
```

Comment (bdraco) [reply Lib/selectors.py]: sounds good

File: Lib/selectors.py [hunk 2]
```diff
@@ -430,6 +430,9 @@ class PollSelector(_PollLikeSelector):
 
 if hasattr(select, 'epoll'):
 
+    _NOT_EPOLLIN = ~select.EPOLLIN
+    NOT_EPOLLOUT = ~select.EPOLLOUT
```
Comment (bdraco) [inline Lib/selectors.py]:
```suggestion
    _NOT_EPOLLOUT = ~select.EPOLLOUT
```

File: Lib/selectors.py [hunk 3]
```diff
@@ -452,23 +455,28 @@ def select(self, timeout=None):
...
+            for fd, event in fd_event_list:
+                key = fd_to_key.get(fd)
                 if key:
-                    ready.append((key, events & key.events))
+                    ready.append(
+                        (
+                            key,
+                            (
+                                (event & NOT_EPOLLIN and EVENT_WRITE)
+                                | (event & NOT_EPOLLOUT and EVENT_READ)
```
Comment (bdraco) [inline Lib/selectors.py]:
```suggestion
                                (event & _NOT_EPOLLIN and EVENT_WRITE)
                                | (event & _NOT_EPOLLOUT and EVENT_READ)
```

**Review by eendebakpt — APPROVED** [review]
The changes look good to me. It took a bit of time for me to understand how the operation of `and` on integers is used in the event processing. The readability of that part of the code is not great. But it is not worse than before, so I am fine with the changes as is.

File: Lib/selectors.py [hunk 4]
```diff
@@ -452,23 +455,28 @@ def select(self, timeout=None):
...
+                    ready.append(
+                        (
+                            key,
+                            (
+                                (event & _NOT_EPOLLIN and EVENT_WRITE)
+                                | (event & _NOT_EPOLLOUT and EVENT_READ)
+                            )
+                            & key.events
+                        )
+                    )
```
Comment (methane) [inline Lib/selectors.py]: I feel this is going too far.
How about splitting this into two lines?

```suggestion
                    events = ((event & _NOT_EPOLLIN and EVENT_WRITE)
                              | (event & _NOT_EPOLLOUT and EVENT_READ))
                    ready.append((key, events & key.events))
```

Comment (bdraco) [reply Lib/selectors.py]: That looks nicer 👍

Comment (bdraco) [reply Lib/selectors.py]: thanks

**Comment by methane** [comment]
Would you do some micro benchmark?

**Comment by methane** [comment]
I missed your benchmark result in the issue.

**Pull request merged**